### PR TITLE
Docs: clean link urls to only use relative paths for Docs links

### PR DESCRIPTION
**Pull request by AlbertCarreras**
Docs: clean link urls to only use relative paths for Docs links

### CODE REVIEW

File: docs/markdown/android/icon.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ fullwidth: true
 - For decorative purposes or visual embellishment, like how you would use an illustration. Contact us if this is needed.
 - As a visual reinforcement for associated text without adding new meaning.
 - To communicate status or health. Use Status instead.
-- As an interactive element (e.g., hover, focus, click/tap). Use [IconButton](https://gestalt.pinterest.systems/web/iconbutton) instead.
+- As an interactive element (e.g., hover, focus, click/tap). Use [IconButton](/web/iconbutton) instead.
```
Comment (dangerismycat) [inline docs/markdown/android/icon.md:23]: This should link to the Android IconButton page

File: docs/markdown/android/textfield.md [hunk 2]
```diff
@@ -16,7 +16,7 @@ fullwidth: true
 </Group>
 <Group>
 <Dont title="When not to use" />
-- Situations where long amounts of text need to be entered, since the full content of the TextField will be truncated. Use [TextArea](https://gestalt.pinterest.systems/web/textarea) instead.
+- Situations where long amounts of text need to be entered, since the full content of the TextField will be truncated. Use [TextArea](/web/textarea) instead.
```
Comment (dangerismycat) [inline docs/markdown/android/textfield.md:19]: This probably shouldn't link at all?

File: docs/markdown/ios/avatar.md [hunk 3]
```diff
@@ -15,7 +15,7 @@ fullwidth: true
   </Group>
   <Group>
   <Dont title="When not to use" />
-     - To represent a group of people, companies and/or brands. use [AvatarGroup](https://gestalt.pinterest.systems/web/avatargroup) instead
+     - To represent a group of people, companies and/or brands. use [AvatarGroup](/web/avatargroup) instead
```
Comment (dangerismycat) [inline docs/markdown/ios/avatar.md:18]: This also probably shouldn't link at all?

File: docs/markdown/ios/icon.md [hunk 4]
```diff
@@ -20,7 +20,7 @@ fullwidth: true
 - For decorative purposes or visual embellishment, like how you would use an illustration. Contact us if this is needed.
 - As a visual reinforcement for associated text without adding new meaning.
 - To communicate status or health. Use Status instead.
-- As an interactive element (e.g., hover, focus, click/tap). Use [IconButton](https://gestalt.pinterest.systems/web/iconbutton) instead.
+- As an interactive element (e.g., hover, focus, click/tap). Use [IconButton](/web/iconbutton) instead.
```
Comment (dangerismycat) [inline docs/markdown/ios/icon.md:23]: This should link to the iOS IconButton

File: docs/markdown/ios/textfield.md [hunk 5]
```diff
@@ -16,7 +16,7 @@ fullwidth: true
 </Group>
 <Group>
 <Dont title="When not to use" />
-- Situations where long amounts of text need to be entered, since the full content of the TextField will be truncated. Use [TextArea](https://gestalt.pinterest.systems/web/textarea) instead.
+- Situations where long amounts of text need to be entered, since the full content of the TextField will be truncated. Use [TextArea](/web/textarea) instead.
```
Comment (dangerismycat) [inline docs/markdown/ios/textfield.md:19]: This probably shouldn't link?

**Review by dangerismycat — APPROVED** [review]
We should also document somewhere what our heuristic is for determining if the path should be included or not (I believe it's only necessary in autogen prop links, but there may be others?)

File: docs/markdown/android/textfield.md (hunk 2 above)
Comment (AlbertCarreras) [reply docs/markdown/android/textfield.md:19]: DONE

File: docs/markdown/ios/icon.md (hunk 4 above)
Comment (AlbertCarreras) [reply docs/markdown/ios/icon.md:23]: DONE

**Pull request merged**